sources/core/Stride.Core/Settings/AppSettingsManager.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using Stride.Core.Reflection;
+
+namespace Stride.Core.Settings
+{
+    /// <summary>
+    /// Manages the loading of application settings with <see cref="IAppSettingsProvider"/>.
+    /// </summary>
+    public static class AppSettingsManager
+    {
+        /// <summary>
+        /// <see cref="AppSettings"/> instance for the application.
+        /// </summary>
+        /// <note>
+        /// Loaded with an <see cref="IAppSettingsProvider"/>. If no provider implementation is found, returns empty <see cref="AppSettings"/> instance.
+        /// </note>
+        public static AppSettings Settings
+        {
+            get
+            {
+                if (settings != null)
+                    return settings;
+
+                foreach (var assembly in AssemblyRegistry.FindAll())
+                    foreach (var type in assembly.GetTypes())
```

Comment:
Listing all types in all assemblies is something we are trying to avoid in the runtime (as opposed to design tools).
Reason is it forces lot of initialization (JIT, cctor, etc.). I remember some cases were it was throwing exceptions on some platforms (maybe not if using `FindAll()`, but at least when listing all referenced assemblies).

The recommended method is to use the assembly processor to list and register those types inheriting from `IAppSettingsProvider`, that way the work is done once at compile time only.
In order to do that, please add `[AssemblyScan]` to `IAppSettingsProvider`. Then there's some logic in `AssemblyRegistry.RegisterEngineAssembly` to register in specific collections (but ideally this should be external for easier extension later)

Reply:
Thanks for the info! I didn't know about `AssemblyScan` and I will definitely also use it in my project. Pushed a fix.